### ISSUE TITLE
Pyright/Pylance misresolves @step overload for list[EvalLog] signatures

### ISSUE THREAD

**Issue by alexandraabbas** [meridianlabs-ai/inspect_flow#666]
## Summary

Pyright / Pylance fails to match the `list[EvalLog]` overload of the `@step` decorator and falls back to the single-`EvalLog` overload, producing false type errors on both user-defined steps and the built-in `tag` / `metadata` / `copy` API functions.

## Reproduction

```python
from inspect_ai.log import EvalLog
from inspect_flow import step
from inspect_flow.api import tag, metadata

# 1. User-defined step — error on the @step decorator itself:
#    Argument 1 to "step" has incompatible type
#    "Callable[[list[EvalLog], str | None, str], list[EvalLog]]";
#    expected "Callable[[EvalLog, str | None, str], EvalLog]"
@step
def qa_auto(
    logs: list[EvalLog],
    model: str | None = None,
    scan_model: str = "openai/gpt-4o",
) -> list[EvalLog]:
    return logs

# 2. Calling tag/metadata — error on the first argument:
#    Argument of type "list[EvalLog]" cannot be assigned to
#    parameter of type "EvalLog | str"
logs: list[EvalLog] = []
tag(logs, add=["foo"])
metadata(logs, set={"key": "value"})
```

## Root cause

The runtime `step()` function (in `_steps/step.py`) is typed as:

```python
def step(func: StepFunction[P]) -> WrappedStepFunction[P]:
```

But the `_StepDecorator` Protocol defines two overloads:

```python
class _StepDecorator(Protocol):
    @overload
    def __call__(
        self, func: Callable[Concatenate[EvalLog, P], EvalLog]
    ) -> Callable[Concatenate[EvalLog | str, P], EvalLog]: ...

    @overload
    def __call__(self, func: StepFunction[P]) -> WrappedStepFunction[P]: ...
```

Pyright appears to match overload 1 (single `EvalLog`) instead of overload 2 (`list[EvalLog]`), and then reports the mismatch. The `_StepDecorator` protocol isn't actually applied to the `step` function at runtime — `step` is typed only with the list overload — so pyright may also just be struggling with `ParamSpec` + `Concatenate` on the direct signature.

The same issue causes false errors when calling `tag()`, `metadata()`, etc. since they are themselves decorated with `@step` and their wrapped signatures aren't resolved correctly by pyright.

## Environment

- **Pylance** (VS Code) / **pyright**
- **inspect_flow**: `0.7.1.dev44` (branch `steps_multiple`, commit `5309bafd`)
- **Python**: 3.12

## Expected behavior

Pyright should resolve the `list[EvalLog]` overload without errors, both for user-defined `@step` functions and for the built-in API functions (`tag`, `metadata`, `copy`).
